editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx

```diff
@@ -126,20 +133,34 @@ export function useMaybeHighlightElement(): {
 
       const alreadyHighlighted = pathsEqual(target, highlightedViews?.[0])
 
-      if (selectionEnabled && !dragging && !resizing && !inserting && !alreadyHighlighted) {
+      if (
+        selectionEnabled &&
+        !dragging &&
+        !resizing &&
+        !inserting &&
+        !alreadyHighlighted &&
+        !isSelectModeWithArea(mode)
+      ) {
         dispatch([setHighlightedView(target)], 'canvas')
       }
     },
-    [dispatch, stateRef],
+    [dispatch, stateRef, mode],
```

Comment:
same tip about turning mode into a useRefEditorState here and in maybeClearHighlightsOnHoverEnd too